pkg/services/container/morph/executor.go

```diff
@@ -274,3 +291,41 @@ func (s *morphExecutor) GetExtendedACL(_ context.Context, body *container.GetExt
 
 	return res, nil
 }
+
+func (s *morphExecutor) validateToken(t *sessionV2.Token, cID *refs.ContainerID, op sessionV2.ContainerSessionVerb) error {
+	c := t.GetBody().GetContext()
+	cc, ok := c.(*sessionV2.ContainerSessionContext)
+	if !ok {
+		return errors.New("session is not container related")
+	}
+
+	if verb := cc.Verb(); verb != op {
+		return fmt.Errorf("wrong container session operation: %s", verb)
+	}
+
+	if cID == nil { // can be nil for PUT
+		return nil
+	}
+
+	if sessionCID := cc.ContainerID().GetValue(); !bytes.Equal(sessionCID, cID.GetValue()) {
+		return fmt.Errorf("wrong container: %s", base58.Encode(sessionCID))
+	}
+
+	var cIDSDK cid.ID
+	err := cIDSDK.ReadFromV2(*cID)
+
+	if err != nil {
+		return fmt.Errorf("cID converting: %w", err)
+	}
+
+	cnr, err := s.rdr.Get(cIDSDK)
+	if err != nil {
+		return fmt.Errorf("reading container from the network: %w", err)
+	}
+
+	if issuer := t.GetBody().GetOwnerID().GetValue(); !bytes.Equal(cnr.Value.Owner().WalletBytes(), issuer) {
+		return fmt.Errorf("session session was issued not by the owner, issuer: %x", issuer)
```

Comment:
```suggestion
		return fmt.Errorf("session was not issued by the container owner %s", cnr.Value.Owner())
```

Reply:
i can understand you but still think issuer is smth that can be lost (and a token itself) while container owner is a constant. also, i would say that errors should highlight mistakes (container owner can not be an error, but incorrect issuer can)

Reply:
> issuer is smth that can be lost

suggest leaving this to the transport security layer

ok, i dont insist, for now this is negligible data